src/amo/commands/updateTaskbar.ts

```diff
@@ -1,16 +1,35 @@
-import fetch from "node-fetch";
+import * as fs from "fs";
 import * as path from "path";
 import * as vscode from "vscode";
 
-import constants from "../../config/config";
+import { addonInfoFromCache } from "../utils/addonCache";
 
 export const statusBarItem = vscode.window.createStatusBarItem(
   vscode.StatusBarAlignment.Left,
   100
 );
 statusBarItem.text = "Assay";
 
-export async function updateTaskbar() {
+export async function findGuidInCache(
```

Comment:
I may be missing something but I'm not a huge fan of this function / the idea behind it. Shouldn't the path `part` that contains the guid always be at a fixed location compared to the `storagePath` ? I feel like you could replace this with a function that extract the guid part from the path - without even bothering to check whether or not a cache file exists.

Reading the directory cache contents looking for each `part` of the path seems like a waste, especially as the cache will likely grow over time.

Reply:
I agree it is a waste. Something andrew and I discussed was to have a way to accurately determine if the guid in the path is the actual guid of an addon. This is for the cases when 

1. A dev nests an addon folder within an addon folder by accident
2. A potential addon version contains a folder with the name of a guid in the cache.

My initial way of extracting the addon from the path required that the guid folder be in the top level of the root of the workspace, but that's too restrictive for devs (they may move folders around, like into `/not-yet-reviewed/` and `/reviewed/`. 

This one here was an attempt at ensuring the guid can be gotten from the path no matter the folder structure. I think the most accurate way of getting the guid from the path (what andrew and I discussed) is that in some folder of the addon, there is a manifest.json file. The lowest level manifest.json in the file tree belongs to the addon of interest, then simply access the folder name 2 levels above. So

`.../.../root/this-is-an-addon-guid/addon-guid-looking-thing/1.0.0/manifest.json` implies that `addon-guid-looking-thing` is the guid. 

Drawbacks of this is an addon can have many nested folders with many files, so looking for a manifest.json as we go up the tree can take a lot of time (since we are using the path of the active file). 

I'm unsure of the most efficient solution for this, maybe a hashtable in the cache that can quickly check if a path part (going reverse order) is a piece of the cache and if so which file it is? Does not handle case 2 though

Reply:
cc @eviljeff for more input.

Reply:
We're going with a fixed directory root that all add-ons will be stored under, which will be settable via an extension setting.